src/main/io/adsb.c

```diff
@@ -0,0 +1,92 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+ 
+
+#include "io/adsb.h"
+#include "navigation/navigation.h"
+#include "common/maths.h"
+#include "math.h"
+#include "common/time.h"
+#include "drivers/time.h"
+
+adsbVehicle_t adsb;
+static timeUs_t lastADSB = 0;
+
+void GPS_distance_cm_bearing(int32_t currentLat1, int32_t currentLon1, int32_t destinationLat2, int32_t destinationLon2, uint32_t *dist, int32_t *bearing)
+{
+  #define DISTANCE_BETWEEN_TWO_LONGITUDE_POINTS_AT_EQUATOR    1.113195f  // MagicEarthNumber from APM
```

Comment:
This should be defined elsewhere. It is already defined in `navigation_private.h`. Perhaps it should be moved in `navigation.h`

Reply:
If OK I would like to leave here - Once I have fully understood the navigation code I  think we can remove and migrate to use existing - but it is quite complex and will take some time..